src/rngs/entropy.rs

```diff
@@ -167,12 +139,19 @@ impl CryptoRng for EntropyRng {}
 
 
 trait EntropySource {
-    fn new_and_fill(dest: &mut [u8]) -> Result<Self, Error>
-        where Self: Sized;
-
+    /// Name of this source
+    fn name() -> &'static str;
+    
+    /// Is this source available?
+    /// 
+    /// The default implementation returns `true`.
+    fn is_available() -> bool { true }
```

Comment:
Should we even provide a default implementation? I feel like this should always be implemented explicitly.